### PR TITLE
*: add print sample config

### PR DESCRIPTION
**Pull request by IANTHEREAL**
### What problem does this PR solve? 
let user easy to find sample config file

### What is changed and how it works?
* add description in README.md
* add `--print-sample-config` in dm-worker and dm-master

### Check List 

Tests 
- Manual test (add detailed scripts or steps below)
```➜  dm git:(ian/master) ✗ bin/dm-master -print-sample-config

# Master Configuration.

// log configuration
log-level = "info"
log-file = "dm-master.log"

// dm-master listen address
master-addr = ":8261"

# replication group <-> dm-Worker deployment, we'll refine it when new deployment function is available
[[deploy]]
source-id = "mysql-replica-01"
dm-worker = "172.16.10.72:8262"

[[deploy]]
source-id = "mysql-replica-02"
dm-worker = "172.16.10.73:8262"
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +35,30 @@ make dmctl      # build dmctl
 
 When DM is built successfully, you can find binaries in the `bin` directory.
 
+## install DM
+
+* the best way to install DM, is via [dm-ansible](https://pingcap.com/docs/tools/data-migration-deployment/)
```
Comment (csuzhangxc) [inline README.md]:
```suggestion
* The best way to install DM is via [DM-Ansible](https://pingcap.com/docs/tools/data-migration-deployment/)
```

File: README.md [hunk 2]
```diff
@@ -35,6 +35,30 @@ make dmctl      # build dmctl
 
 When DM is built successfully, you can find binaries in the `bin` directory.
 
+## install DM
```
Comment (csuzhangxc) [inline README.md]: `Install DM`? or `Installing`?

File: README.md [hunk 3]
```diff
@@ -35,6 +35,30 @@ make dmctl      # build dmctl
...
+
+* the best way to install DM, is via [dm-ansible](https://pingcap.com/docs/tools/data-migration-deployment/)
+* deploy DM manually
+```
```
Comment (csuzhangxc) [inline README.md]: adding indentation to indicate is a manual method.

File: README.md [hunk 4]
```diff
@@ -35,6 +35,30 @@ make dmctl      # build dmctl
...
+sha256sum -c dm-latest-linux-amd64.sha256
+
+# Extract the package.
+tar -xzf dm-latest-linux-amd64.tar.gz
+cd dm-latest-linux-amd64
+```
+
+## Config File
+
+* all sample config files can be found in conf directory of dm tarball
```
Comment (csuzhangxc) [inline README.md]: whether needing to wrap conf as `conf`? if so, do it for lines below.

File: dm/master/config.go [hunk 5]
```diff
@@ -96,6 +105,20 @@ func (c *Config) Parse(arguments []string) error {
 		return flag.ErrHelp
 	}
 
+	if c.printSampleConfig {
+		if strings.TrimSpace(SampleConfigFile) == "" {
+			fmt.Println("sample config file of dm-master is empty")
+		} else {
+			rawConfig, err := base64.StdEncoding.DecodeString(SampleConfigFile)
+			if err != nil {
+				fmt.Println("base64 decode config error:", err)
```
Comment (csuzhangxc) [inline dm/master/config.go]: print `SampleConfigFile`?

Comment (GregoryIan) [reply dm/master/config.go]: no need, only need to build a new one

File: dm/master/dm-master.toml [hunk 6]
```diff
@@ -1,13 +1,13 @@
 # Master Configuration.
 
+// log configuration
```
Comment (amyangfei) [inline dm/master/dm-master.toml]: `toml` file use `#` as comment leading?

Comment (GregoryIan) [reply dm/master/dm-master.toml]: yep I would fix it

File: Makefile [hunk 7]
```diff
@@ -21,9 +20,19 @@ ifeq ("$(WITH_RACE)", "1")
 	GOBUILD   = CGO_ENABLED=1 $(GO) build
 endif
 
-ARCH      := "`uname -s`"
+ARCH      := "$(shell uname -s)"
 LINUX     := "Linux"
 MAC       := "Darwin"
+SAMPLE_WORK_CONFIG := ""
+SAMPLE_WORK_CONFIG := ""
```
Comment (csuzhangxc) [inline Makefile]: is this needed?

Comment (GregoryIan) [reply Makefile]: forget to remove it

**Comment by csuzhangxc** [comment]
LGTM

**Pull request merged**